subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/writer/WriteDependencyVerificationFile.java

```diff
@@ -578,7 +603,7 @@ private static boolean hasAtLeastOnePublicKey(PGPPublicKeyRing ring) {
         return ring.getPublicKeys().hasNext();
     }
 
-    private List<PGPPublicKeyRing> loadExistingKeyRing(BuildTreeDefinedKeys keyrings) throws IOException {
+    protected static List<PGPPublicKeyRing> loadExistingKeyRing(BuildTreeDefinedKeys keyrings, boolean isDryRun) throws IOException {
         List<PGPPublicKeyRing> existingRings;
         if (!isDryRun) {
             existingRings = keyrings.loadKeys();
```

Comment:
A question so I better understand the change: so `keyrings .loadKeys()` would load all keys from a file to one keyring right now?

Reply:
Yes, it will load a predefined keyring: https://github.com/gradle/gradle/blob/47bff9d4dee8352f8ebb6a1c73c7ac7ffd72c326/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/verification/signatures/BuildTreeDefinedKeys.java#L63-L69